cocotb/share/lib/utils/cocotb_utils.c

```diff
@@ -66,7 +66,17 @@ void* utils_dyn_open(const char* lib_name)
     SetErrorMode(0);
     ret = LoadLibrary(lib_name);
     if (!ret) {
-        printf("Unable to open lib %s\n", lib_name);
+        printf("Unable to open lib %s", lib_name);
+        LPSTR msg_ptr;
+        if (FormatMessageA(FORMAT_MESSAGE_FROM_SYSTEM |
+                           FORMAT_MESSAGE_ALLOCATE_BUFFER, NULL,
+                           GetLastError(),
+                           MAKELANGID(LANG_NEUTRAL, SUBLANG_SYS_DEFAULT),
+                           (LPSTR)&msg_ptr, 255, NULL)) {
```

Comment:
This looks very wrong to me - casting a `T*` to a `T` is nonsense.

Reply:
Huh, guess the windows API is just bizarre: https://stackoverflow.com/a/24214707/102441

Reply:
I took it directly from the docs: https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-formatmessagea